src/main/java/uk/gov/pay/directdebit/events/exception/EventHasNoPaymentIdException.java

```diff
@@ -0,0 +1,12 @@
+package uk.gov.pay.directdebit.events.exception;
+
+import uk.gov.pay.directdebit.common.exception.NotFoundException;
+
+import static java.lang.String.format;
+
+public class EventHasNoPaymentIdException extends NotFoundException {
```

Comment:
It's more specific and shows that the Event itself has no payment id on it, and is the problem, rather than simply not finding something in the database. We'd like to keep it as-is if you don't mind.